io/io/src/TBufferMerger.cxx

```diff
@@ -54,10 +54,11 @@ TBufferMerger::~TBufferMerger()
 
 std::shared_ptr<TBufferMergerFile> TBufferMerger::GetFile()
 {
-   R__LOCKGUARD(gROOTMutex);
    std::shared_ptr<TBufferMergerFile> f(new TBufferMergerFile(*this));
-   gROOT->GetListOfFiles()->Remove(f.get());
-   fAttachedFiles.push_back(f);
+   {
+      R__LOCKGUARD(gROOTMutex); // This mutex can be replaced by a local one?
```

Comment:
Since fAttachedFiles is a simple vector, yes the lock/mutex could be a local one (used here and at all access to fAttachedFiles).

Reply:
Thanks Philippe for the comments. 
We are running benchmarks. This is surely the last mile.